From 6ad37d89ab2fd1bf32b1f74a9c81bc4a4384b8ac Mon Sep 17 00:00:00 2001
From: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Date: Fri, 4 Aug 2023 15:01:49 +0200
Subject: [PATCH 0573/1141] ARM: dts: stm32: populate all timer counter nodes
 on stm32mp15

Counter driver originally had support limited to quadrature interface and
simple counter. It has evolved, so add remaining counter nodes in stm32
timer nodes.

Signed-off-by: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Change-Id: I626261f863dee73f88e4b35b4524fa6939dc8dc2
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/323453
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 arch/arm/boot/dts/stm32mp151.dtsi | 41 +++++++++++++++++++++++++++++++
 1 file changed, 41 insertions(+)

--- a/arch/arm/boot/dts/stm32mp151.dtsi
+++ b/arch/arm/boot/dts/stm32mp151.dtsi
@@ -647,6 +647,11 @@
 				feature-domains = <&etzpc STM32MP1_ETZPC_TIM6_ID>;
 				status = "disabled";
 
+				counter {
+					compatible = "st,stm32-timer-counter";
+					status = "disabled";
+				};
+
 				timer@5 {
 					compatible = "st,stm32h7-timer-trigger";
 					reg = <5>;
@@ -668,6 +673,11 @@
 				feature-domains = <&etzpc STM32MP1_ETZPC_TIM7_ID>;
 				status = "disabled";
 
+				counter {
+					compatible = "st,stm32-timer-counter";
+					status = "disabled";
+				};
+
 				timer@6 {
 					compatible = "st,stm32h7-timer-trigger";
 					reg = <6>;
@@ -687,6 +697,11 @@
 				feature-domains = <&etzpc STM32MP1_ETZPC_TIM12_ID>;
 				status = "disabled";
 
+				counter {
+					compatible = "st,stm32-timer-counter";
+					status = "disabled";
+				};
+
 				pwm {
 					compatible = "st,stm32-pwm";
 					#pwm-cells = <3>;
@@ -712,6 +727,11 @@
 				feature-domains = <&etzpc STM32MP1_ETZPC_TIM13_ID>;
 				status = "disabled";
 
+				counter {
+					compatible = "st,stm32-timer-counter";
+					status = "disabled";
+				};
+
 				pwm {
 					compatible = "st,stm32-pwm";
 					#pwm-cells = <3>;
@@ -737,6 +757,11 @@
 				feature-domains = <&etzpc STM32MP1_ETZPC_TIM14_ID>;
 				status = "disabled";
 
+				counter {
+					compatible = "st,stm32-timer-counter";
+					status = "disabled";
+				};
+
 				pwm {
 					compatible = "st,stm32-pwm";
 					#pwm-cells = <3>;
@@ -1214,6 +1239,11 @@
 				feature-domains = <&etzpc STM32MP1_ETZPC_TIM15_ID>;
 				status = "disabled";
 
+				counter {
+					compatible = "st,stm32-timer-counter";
+					status = "disabled";
+				};
+
 				pwm {
 					compatible = "st,stm32-pwm";
 					#pwm-cells = <3>;
@@ -1242,11 +1272,17 @@
 				feature-domains = <&etzpc STM32MP1_ETZPC_TIM16_ID>;
 				status = "disabled";
 
+				counter {
+					compatible = "st,stm32-timer-counter";
+					status = "disabled";
+				};
+
 				pwm {
 					compatible = "st,stm32-pwm";
 					#pwm-cells = <3>;
 					status = "disabled";
 				};
+
 				timer@15 {
 					compatible = "st,stm32h7-timer-trigger";
 					reg = <15>;
@@ -1269,6 +1305,11 @@
 				feature-domains = <&etzpc STM32MP1_ETZPC_TIM17_ID>;
 				status = "disabled";
 
+				counter {
+					compatible = "st,stm32-timer-counter";
+					status = "disabled";
+				};
+
 				pwm {
 					compatible = "st,stm32-pwm";
 					#pwm-cells = <3>;
